test/ntttcp_test.py

```diff
@@ -0,0 +1,441 @@
+        assert parse_result.is_bandwidth_and_rate_limit(cmd_bandwidth, bandwidth_limit) is True
+
+    def test_output_log_file(self) -> None:
+        common_option = "-O"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = ParseNtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_output_log_file()
+        throughput = parse_result.get_throughput()
+        print("throughput=", throughput)
+#       #assert throughput >= self.expected_throughput * 0.9
+
+    def test_output_xml_file(self) -> None:
+        common_option = "-x"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = ParseNtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_output_xml_file()
+        throughput = parse_result.get_throughput()
+        print("throughput=", throughput)
+#       assert throughput >= self.expected_throughput * 0.9
+
+    def test_ouput_json_file(self) -> None:
+        common_option = "-j"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = ParseNtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_output_json_file()
+        throughput = parse_result.get_throughput()
+        print("throughput=", throughput)
+#        assert throughput >= self.expected_throughput * 0.9
+
+    def test_starting_port_number(self) -> None:
+        common_option = "-p 10000"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = ParseNtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_starting_port_number(send_cmd) is True
+
+    def test_mapping_option(self) -> None:
+        rec_cmd = "ulimit -n 10240 && ntttcp -D -r -m 200,*,127.0.0.1 -D -t 5 -V"
+        send_cmd = "ulimit -n 10240 && ntttcp -s127.0.0.1 -P 200 -t 5"
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = ParseNtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_mapping_option(rec_cmd) is True
+
+    def test_system_tcp_retransmit(self) -> None:
+        common_option = "--show-tcp-retrans"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = ParseNtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_system_tcp_retransmit() is True
+
+    def test_show_nic_packets(self) -> None:
+        common_option = "--show-nic-packets eth0"
```

Comment:
If we use loopback interface 127.0.0.1, we couldn't use eth0, there is no packet stream through this interface, we should use "lo".

=> ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.1.69.4  netmask 255.255.255.0  broadcast 10.1.69.255
        inet6 fe80::20d:3aff:fefe:f2a8  prefixlen 64  scopeid 0x20<link>
        ether 00:0d:3a:fe:f2:a8  txqueuelen 1000  (Ethernet)
        RX packets 1460676  bytes 715608521 (715.6 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1453507  bytes 377067657 (377.0 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

**lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0**
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 997601321  bytes 41843846412373 (41.8 TB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 997601321  bytes 41843846412373 (41.8 TB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

Reply:
done